### PR TITLE
Added PSA testing and DB Create & Delete helper methods

### PR DESCRIPTION
**Pull request by Lindessa**

### CODE REVIEW

File: Source/Test/Microsoft.Deployment.Site.Test/Microsoft-PSATemplateTests.cs [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            var acceptButton = driver.FindElementById("cred_accept_button");
+            acceptButton.Click(); 
+
+            Thread.Sleep(new TimeSpan(0, 0, 2));
+
+            var validated = driver.FindElementByClassName("st-validated");
+            Assert.IsTrue(validated.Text == "Successfully validated");
+        }
+
+        public void Given_CorrectSearchTerms_When_Validating_Then_Success()
```
Comment (cdoroftei) [inline Source/Test/Microsoft.Deployment.Site.Test/Microsoft-PSATemplateTests.cs]: probably not needed :)

File: Source/Test/Microsoft.Deployment.Site.Test/Microsoft.Deployment.Site.Test.csproj [hunk 2]
```diff
@@ -146,6 +149,12 @@
     </Content>
     <Content Include="Readme.txt" />
   </ItemGroup>
+  <ItemGroup>
```
Comment (cdoroftei) [inline Source/Test/Microsoft.Deployment.Site.Test/Microsoft.Deployment.Site.Test.csproj]: why was this included?

File: Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/HelperMethods.cs [hunk 3]
```diff
@@ -294,5 +370,61 @@ public static void CleanSubscription(string username, string password, string te
 
             client.ResourceGroups.Delete(resourceGroupName);
         }
+
+        public static void CreateDatabase(string server, string username, string password, string database)
+        {
+            string sqlConnectionString = getSqlConnectionString(server, username, password, "master");
+            string script = "CREATE DATABASE " + database;
+
+            SqlConnection connection = new SqlConnection(sqlConnectionString);
```
Comment (raymondb1) [inline Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/HelperMethods.cs]: Use using for objects implementing IDisposable

File: Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/HelperMethods.cs [hunk 4]
```diff
@@ -294,5 +370,61 @@ public static void CleanSubscription(string username, string password, string te
...
+                }
+            }
+        }
+   
+
+
+
+        public static string getSqlConnectionString(string server, string username, string password, string database)
+        {
+            string result = "Data Source=" + server + ";Initial Catalog=" + database + ";Integrated Security=False;" +
```
Comment (raymondb1) [inline Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/HelperMethods.cs]: Use interpolated string $"Data Source={server}", etc.

File: Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/PSACreds.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Deployment.Site.Test
```
Comment (raymondb1) [inline Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/PSACreds.cs]: Make them properties only if it makes sense or other APIs can't serialize class members (only properties)

File: Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/HelperMethods.cs [hunk 6]
```diff
@@ -294,5 +370,61 @@ public static void CleanSubscription(string username, string password, string te
 
             client.ResourceGroups.Delete(resourceGroupName);
         }
+
+        public static void CreateDatabase(string server, string username, string password, string database)
+        {
+            string sqlConnectionString = getSqlConnectionString(server, username, password, "master");
+            string script = "CREATE DATABASE " + database;
```
Comment (raymondb1) [inline Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/HelperMethods.cs]: Create database should specify service level S1 and Latin1 general 100 collation
Similar to:
CREATE DATABASE <Database_Name, sysname, Database_Name> COLLATE <collation_Name, sysname, SQL_Latin1_General_CP1_CI_AS> 
	(
	  EDITION = '<EDITION, , Standard>',
	  SERVICE_OBJECTIVE='<SERVICE_OBJECTIVE,,S0>',
	  MAXSIZE = <MAX_SIZE,,1024 GB>
	)

File: Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/HelperMethods.cs (hunk 4 above)
Comment (Lindessa) [inline Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/HelperMethods.cs]: done

File: Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/PSACreds.cs (hunk 5 above)
Comment (Lindessa) [inline Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/PSACreds.cs]: done

File: Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/HelperMethods.cs (hunk 6 above)
Comment (Lindessa) [inline Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/HelperMethods.cs]: Using 
"CREATE DATABASE " + database +
 " COLLATE Latin1_General_100_CI_AS ( EDITION = 'Standard', SERVICE_OBJECTIVE = 'S1' )";

File: Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/HelperMethods.cs (hunk 3 above)
Comment (Lindessa) [inline Source/Test/Microsoft.Deployment.Site.Test/TestHelpers/HelperMethods.cs]: done

File: Source/Test/Microsoft.Deployment.Site.Test/Microsoft-PSATemplateTests.cs (hunk 1 above)
Comment (Lindessa) [inline Source/Test/Microsoft.Deployment.Site.Test/Microsoft-PSATemplateTests.cs]: Removed!

**Comment by Lindessa** [comment]
https://github.com/Microsoft/BusinessPlatformApps/pull/588
